src/main/java/com/example/petbuddybackend/entity/user/Caretaker.java

```diff
@@ -0,0 +1,38 @@
+package com.example.petbuddybackend.entity.user;
+
+import com.example.petbuddybackend.entity.address.Address;
+import com.example.petbuddybackend.entity.animal.AnimalType;
+import jakarta.persistence.*;
+import lombok.*;
+
+import java.util.Set;
+
+@Entity
+@Builder
+@Getter @Setter
+@AllArgsConstructor
+@NoArgsConstructor
+public class Caretaker {
+
+    @Id
+    private String email;
+
+    @Column(nullable = false, length = 14)
+    private String phoneNumber;
+
+    @Column(length = 1500)
+    private String description;
+
+    @Enumerated(EnumType.STRING)
+    @ElementCollection(targetClass = AnimalType.class, fetch = FetchType.EAGER)
+    @JoinTable(name = "animals_taken_care_of", joinColumns = @JoinColumn(name = "email"))
+    @Column(name = "takes_care_of", nullable = false)
+    Set<AnimalType> animalsTakenCareOf;
+
+    @OneToOne(cascade = CascadeType.ALL, optional = false)
+    private Address address;
+
+    @OneToOne(cascade = { CascadeType.MERGE, CascadeType.REMOVE }, optional = false)
+    @PrimaryKeyJoinColumn
+    private AppUser accountData;
```

Comment:
raczej bym tutaj kaskady nie wprowadzal, a na pewno nie REMOVE.
Konto uzytkownika zostanie zalozone wraz z jego pierwsza interakcja z serwisem, wiec nic nie trzeba bedzie dorabiac przy tworzeniu Caretaker, a jesli usuniemy Caretaker, to z AppUser moze byc polaczony Customer

Reply:
Faktycznie zła strona relacji ma ten Cascade.REMOVE. W zamian dodałem to do AppUser żeby po usunięciu AppUsera usunęło Caretaker